### PR TITLE
SEACAS: Test fixes and clang-format

### PR DESCRIPTION
**Pull request by gsjaardema**
@trilinos/seacas 

## Motivation

* Clang format run over most C/C++ files
* Fix most tests that were diffing stdout/stderr to not do that anymore.  Should fix all issues with jsrun or mpiexec or OS 
  adding extraneous junk to stdout/stderr. 
* Fix 1-D array interpolation in aprepro

### CODE REVIEW

**Comment by e10harvey** [comment]
@gsjaardema: Can you please point me to the changes related to:
- Fix most tests that were diffing stdout/stderr to not do that anymore. Should fix all issues with jsrun or mpiexec or OS
adding extraneous junk to stdout/stderr.
- Fix 1-D array interpolation in aprepro

**Comment by gsjaardema** [comment]
The test changes are in
packages/SEACAS/libraries/ioss/src/main/CMakeLists.txt

The aprepro changes are in apr_aprepro.cc

..Greg

**Comment by gsjaardema** [comment]
@e10harvey Did you need any further information about where the specific changes were?

File: packages/seacas/libraries/aprepro_lib/apr_aprepro.cc [hunk 1]
```diff
@@ -24,10 +24,24 @@
 
 namespace {
   const unsigned int HASHSIZE       = 5939;
-  const char *       version_string = "5.27 (2021/07/20)";
+  const char *       version_string = "5.29 (2021/07/22";
```
Comment (e10harvey) [inline packages/seacas/libraries/aprepro_lib/apr_aprepro.cc:27]:
```suggestion
  const char *       version_string = "5.29 (2021/07/22)";
```

File: packages/seacas/libraries/ioss/src/main/CMakeLists.txt [hunk 2]
```diff
@@ -251,48 +235,37 @@ else()
...
 TRIBITS_ADD_ADVANCED_TEST(exodus64_to_unstructured_cgns
    TEST_0 EXEC io_shell ARGS ${DECOMP_ARG} --64 ${CMAKE_CURRENT_SOURCE_DIR}/test/8-block.g 8-block64.cgns
      NOEXEPREFIX NOEXESUFFIX
      NUM_MPI_PROCS ${NPROCS}
-   TEST_1 CMND ${CGNS_CGNSLIST_BINARY} ARGS -ld 8-block64.cgns
-     OUTPUT_FILE exodus64_to_unstructured_cgns.tmp
-   TEST_2 CMND grep ARGS -v Information exodus64_to_unstructured_cgns.tmp OUTPUT_FILE exodus64_to_unstructured_cgns.out
-   TEST_3 CMND diff ARGS ${CMAKE_CURRENT_SOURCE_DIR}/test/exodus_to_unstructured_cgns.gold exodus64_to_unstructured_cgns.out
+   TEST_1 CMND ${CGNS_CGNSDIFF_BINARY} ARGS -d 8-block64.cgns ${CMAKE_CURRENT_SOURCE_DIR}/test/8-block.cgns.gold
+     NUM_MPI_PROCS 1
```
Comment (e10harvey) [inline packages/seacas/libraries/ioss/src/main/CMakeLists.txt:244]: Is the following understanding correct?

1. `io_shell` runs as an MPI process and generates `8-block64.cgns`; the test fails before diffing if `io_shell` returns non-zero
2. After 1. is complete, test correctness is checked by diffing `8-block64.cgns` with `8-block.cgns.gold`

**Review by e10harvey — COMMENTED** [review]
@gsjaardema: In general, everything looks great. Going forward, it would help me if formatting changes were in a separate PR or only applied to the code changes. Please see a few comments and questions below.

File: packages/seacas/libraries/ioss/src/main/CMakeLists.txt (hunk 2 above)
Comment (gsjaardema) [reply packages/seacas/libraries/ioss/src/main/CMakeLists.txt:244]: Yes, that is correct understanding

**Comment by gsjaardema** [comment]
> @gsjaardema: In general, everything looks great. Going forward, it would help me if formatting changes were in a separate PR or only applied to the code changes. Please see a few comments and questions below.

Yes, understand, but SEACAS is currently developed externally from Trilinos and only snapshotted in periodically.  It is difficult to separate formatting changes and code changes when only snappshotting every month...

**Comment by gsjaardema** [comment]
@prwolfe @ccober6 @e10harvey

I think this fell off of the radar.  Could this be approved so I can verify that it fixes some of the long-term test failures we have been seeing on the ATDM / HPC platforms.

**Review by prwolfe — APPROVED** [review]

**Pull request merged**